### PR TITLE
Ruby: Fix performance problem in `Definitions.ql`

### PR DESCRIPTION
**Pull request by hvitved**
Avoids computing this large set:

```
[2021-11-12 14:31:11] (542s) Tuple counts for Definitions::definitionOf#ff#min_term/5@cc3ffc after 2m10s:
                      337397265 ~3%     {5} r1 = JOIN Definitions::definitionOf#ff#shared WITH Module::Cached::resolveConstant#ff_10#join_rhs ON FIRST 1 OUTPUT Rhs.1 'arg0', Lhs.2 'arg3', Lhs.1 'arg4', Lhs.2 'arg3', Lhs.1 'arg4'
                                        return r1
```

### CODE REVIEW

File: ruby/ql/src/queries/analysis/Definitions.ql [hunk 1]
```diff
@@ -65,17 +65,19 @@ newtype DefLoc =
     not exists(MethodBase m | m.getAChild+() = write)
   }
 
+pragma[noinline]
+ConstantWriteAccess definitionOf0(string fqn) {
+  fqn = resolveConstant(_) and
+  result =
+    min(ConstantWriteAccess w | w.getQualifiedName() = fqn | w order by w.getLocation().toString())
```
Comment (aibaars) [inline ruby/ql/src/queries/analysis/Definitions.ql:72]: Ordering by `toString` feels a bit brittle. Wouldn't it be better to order by path, startline, startcolumn, etc.?

**Review by aibaars — APPROVED** [review]
Looks good to me. Just one question.

File: ruby/ql/src/queries/analysis/Definitions.ql (hunk 1 above)
Comment (hvitved) [reply ruby/ql/src/queries/analysis/Definitions.ql:72]: This bit was simply moved over from the old implementation.

Comment (hvitved) [reply ruby/ql/src/queries/analysis/Definitions.ql]: I have changed it.

**Review by aibaars — APPROVED** [review]

**Pull request merged**